### PR TITLE
promote PDBUnhealthyPodEvictionPolicy to GA

### PR DESCRIPTION
**Pull request by atiratree**
#### Which issue(s) this PR fixes:

tracking issue: https://github.com/kubernetes/enhancements/issues/3017

#### Does this PR introduce a user-facing change?

```release-note
The PodDisruptionBudget `spec.unhealthyPodEvictionPolicy` field has graduated to GA. This field may be set to `AlwaysAllow` to always allow unhealthy pods covered by the PodDisruptionBudget to be evicted.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3017-pod-healthy-policy-for-pdb
```

### CODE REVIEW

**Comment by pacoxu** [comment]
/kind feature
/priority important-soon

The KEP update was merged.
Please remove the `WIP: ` from the title once this PR is ready for review.

**Comment by atiratree** [comment]
added e2e tests

**Comment by atiratree** [comment]
the PR should be ready for a review

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -1086,7 +1087,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	NodeSwap: {Default: true, PreRelease: featuregate.Beta},
 
-	PDBUnhealthyPodEvictionPolicy: {Default: true, PreRelease: featuregate.Beta},
+	PDBUnhealthyPodEvictionPolicy: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.32
```
Comment (soltysh) [inline pkg/features/kube_features.go:1090]: Nit, remove is usually n+2, so in your case that should be 1.33

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
from sig-apps pov

**Comment by atiratree** [comment]
@liggitt can you please review the promotion?

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**